### PR TITLE
adds comment functionality

### PR DESCRIPTION
**Pull request by CjMoore**

### CODE REVIEW

File: spec/features/comments/user_creates_new_comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe "user fills out comment form on job page" do
```
Comment (CjMoore) [inline spec/features/comments/user_creates_new_comment_spec.rb:3]: Is this the only kind of feature test I need for comments? @s-espinosa

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class CommentsController < ApplicationController
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.job_id = params[:job_id]
+    @comment.save
```
Comment (CjMoore) [inline app/controllers/comments_controller.rb:6]: Also I have validations for comment :author and :content, but I was having trouble getting a conditional to re-render the job show page if those validations weren't met. I tried render :'job/show', but that didn't work. I realize I need to provide some kind of identifier for which job show to render but I'm unsure of the syntax I need to use. Any advice?, or is it not necessary? @s-espinosa

File: spec/features/comments/user_creates_new_comment_spec.rb (hunk 1 above)
Comment (s-espinosa) [reply spec/features/comments/user_creates_new_comment_spec.rb:3]: This looks great. Just two notes:

* Based on the spec I would probably add a test to see that new comments render in the proper order. Take a look at `nth-of-type` and `nth-child` CSS selectors for that test.
* Consistent with the SO article Andrew posted, would recommend using `expect(page).to have_content("Cool job")` over `.has_content?`

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (s-espinosa) [reply app/controllers/comments_controller.rb:6]: That argument that you're passing to render is looking for a specific view template, so the path needs to match the path in your views folder (i.e. switch `:'job/show'` to `:'jobs/show'`).

One additional note: would probably switch the syntax on lines 4-5 to the following:

```ruby
@job     = Job.find(params[:job_id])
@comment = @job.comments.new(comment_params)
```

Has the added advantage of defining `@job` for use in the `jobs/show` view that you're hoping to render on fail.

**Review by s-espinosa — COMMENTED** [review]
@CjMoore Like your tests! Responses to your questions below.

**Pull request merged**